src/type_id.rs

```diff
@@ -157,6 +157,8 @@ pub enum TypeIdPrimitive {
 	U64,
 	/// `u128`
 	U128,
+	/// no rust equivalent
+	U256,
```

Comment:
I not sure about this change. What we try to do with the `type-metadata` crate is to have a file format to communicate the structure of some SCALE encoded values. Since the main implementation of SCALE is in Rust there are of course proper well defined SCALE encodings for all of Rust's integer types, however I am not sure about a `u256` or `i256` type in that regard. In the worst case we'd need to introduce that to SCALE codec or represent it via custom repr of a vector of bytes or so.

Reply:
I see what you mean about parity-scale-codec. However:
- Polkadot supports i256/u256 fine today
- Solang does not depend on parity-scale-codec, it uses its own encoder/decoder
- In Solidity `int` and `uint` default to 256 bits. This is a very common type in Solidity and it would be nice if it could be encoded/decoded with a few wasm instructions

Today Solang has a super-ugly metadata encoder which generates u256 and i256. This works already when using polkadot for sending/receiving transactions.

Now I do agree 256 bits is a very odd size and a bit ugly.

Reply:
I see so this is about what the `type-metadata` crate is actually abstracting. If you want to interact with the already existing infrastructure of Polkadot JS UI you should definitely mimic the SCALE codec. We can definitely add the 256-bit integers as done in this PR but I want to have feedback from the Polkadot JS UI team first.

Reply:
JS API supports U256.